src/utils/setUserScopePermission.js

```diff
@@ -0,0 +1,66 @@
+const { hasWrite } = require('./permissions');
+const { isPaginated } = require('./pagination');
+
+const KEY_NAME = 'scopePermission'; // is forced by this utils
+
+/**
+ * For every request it should return also the user scope, forced over this util.
+ * @param {ResultObject} result The request result, that should modified.
+ * @param {String ||  Array::Permission} permissions
+ * @param {String} permissionsAsString It is set directly for example in create request 'write'.
+ * @param {Array::Permission} permissionsAsArrayOfObjects To check the user scoped permissions and set it.
+ * @param {User} user The user object to set the permission scope.
+ * @returns {ResultObject}
+ */
+const setUserScopePermission = (result, permissions, user) => {
```

Comment:
Is it importent to set result here or is it enoug to give the permission back. Reduce the magic

Reply:
See line 4 the key is forced by this util. It should not set by the services at self.